Sources/Security/Signing.swift

```diff
@@ -259,7 +240,65 @@ extension Signing.SignatureParameters {
 
 private final class BundleToken: NSObject {}
 
-// MARK: - Data extensions
+private extension Signing {
+
+    @available(iOS 13.0, macOS 10.15, watchOS 6.0, tvOS 13.0, *)
+    static func createPublicKey(with data: Data) throws -> PublicKey {
+        return try Algorithm(rawRepresentation: data)
+    }
+
+    static func extractAndVerifyIntermediateKey(
+        from signature: Data,
+        publicKey: Signing.PublicKey
+    ) -> Signing.PublicKey? {
+        guard #available(iOS 13.0, macOS 10.15, watchOS 6.0, tvOS 13.0, *) else { return nil }
+
+        let intermediatePublicKey = signature.component(.intermediatePublicKey)
+        let intermediateKeyExpiration = signature.component(.intermediateKeyExpiration)
+        let intermediateKeySignature = signature.component(.intermediateKeySignature)
+
+        guard publicKey.isValidSignature(intermediateKeySignature,
+                                         for: intermediateKeyExpiration + intermediatePublicKey) else {
+            Logger.warn(Strings.signing.intermediate_key_failed_verification(signature: intermediateKeySignature))
+            return nil
+        }
+
+        guard let expirationDate = Self.extractAndVerifyIntermediateKeyExpiration(intermediateKeyExpiration) else {
+            return nil
```

Comment:
So if the intermediate key is expired, we assume it's an attacker correct? As long as the backend makes sure this doesn't happen, this should be fine 👍

Reply:
Yup /cc @bisho